src/components/GeocodeContext.tsx

```diff
@@ -1,21 +1,21 @@
 import { createContext, useContext } from 'react';
 import { useImmerReducer } from 'use-immer';
 
-const GeocodeContext = createContext();
-
 const initialValues = {
   apiKey: '',
   data: {
     street: '',
     zone: '',
-    file: null,
-    fieldsFromFile: [],
+    file: null as File | null,
+    fieldsFromFile: [] as string[],
     sampleData: {},
     totalRecords: 0,
-    valid: null,
+    valid: null as boolean | null,
   },
 };
 
+const GeocodeContext = createContext(initialValues);
```

Comment:
```suggestion
const GeocodeContext = createContext();
```

Passing `initialValues` to this function seems misleading to me since it doesn't match the shape of the value set in the provider below (`{ geocodeContext: state, geocodeDispatch: dispatch }`)